docs/planning/sprintplanning.md

```diff
@@ -0,0 +1,66 @@
+# Knative Eventing sprint planning
+
+This document summarizes project planning mechanics for Knative Eventing releases. Knative Eventing release is planned every 6 weeks and will continue to be so. Current proposal is to align sprint with the release cycle of 6 weeks. Eventually this may change as we learn from the process.
+
+## How to manage Bugs?
```

Comment:
You dive into "Bugs" here -- it might be worth explaining bug vs story/feature above this point.

I'm wondering whether "bugs" should be in this document. My preference would be to address most bugs in the same release cycle as they are reported, rather than having users wait E(time-to-fix) = 10 weeks (1 week to fix, 6 week release cycle, 0-6 weeks to reach the next planning cycle). Fixing in the same cycle where possible will mean that we won't be able to accomplish as many forward-looking tasks, but I think our users will overall prefer those results.

Reply:
Will summarize what we discussed in-person:
 - Bugs could be customer reported or engineering team reported. We can have separate labels such as CRI or CRB or just customer reported for a customer reported incident or bug.
- Bugs need to be triaged and assigned priority - P0, P1 or P2 (We dont need P3 and P4 or severities)
- P0 - Bugs needs to be fixed in current release and cannot wait for next release
- P1 - Bugs will be planned to be fixed in next release
- P2 - Bugs that need to be fixed and will be done if we have time. We shouldn't use this as a parking for all bugs and should make the hard call and reject/ice-box or close a bug if we think we will not fix it in next 2-3 releases.

There is one more priority which means drop everything and fix it for a hotfix release. We'll get to this when the time comes :)

Fixing all bugs in the same release will be an overkill. However fixing all P1s in the next release will help keep the quality bar high enough that it doesn't affect feature work.

However all of this depends on we being critical when assigning priorities. We can figure out the mechanics as we start triaging.

Reply:
I'd move priorities by one, such that P0 goes into the past release, P1 goes into the in-progress release, and P2 is evaluated for the next release.

When the planning time comes, some number of P2s will be selected and converted to P1s because they are targeted for the next release. The rest will remain as P2s, but will have been looked at.

Reply:
Akash pointed out that the P2s for the next release can simply be tagged with the milestone, rather than needing to have the priority changed.

For now, we can just do this with automation (e.g. using a tool like [`hub`](https://github.com/github/hub)) and add the milestone to all open P2s at the start of the release.